pkgs/applications/virtualization/singularity/packages.nix

```diff
@@ -0,0 +1,50 @@
+{ callPackage
+, fetchFromGitHub
+}:
+
+{
+  apptainer = callPackage (callPackage ./generic.nix rec {
+    pname = "apptainer";
+    version = "1.0.2";
+    projectName = "apptainer";
+
+    src = fetchFromGitHub {
+      owner = "apptainer";
+      repo = "apptainer";
+      rev = "v${version}";
+      sha256 = "3yN2bDpgeOtCzFFhtPLf43wwzDNYxPCFnbgcL/1Mweg=";
+    };
+
+    # Update by running
+    # nix-prefetch -E "{ sha256 }: ((import ./. { }).apptainer.override { vendorSha256 = sha256; }).go-modules"
+    # at the root directory of the Nixpkgs repository
+    vendorSha256 = "N0vbQToRCASKvNW/1bSLUXG2uwwg+0FO4lT3kG3ab04=";
+
+    # This would soon be false
+    # https://github.com/apptainer/apptainer/pull/495
+    defaultToSuid = true;
+
+    extraDescription = " (previously known as Singularity)";
+    extraMeta.homepage = "https://apptainer.org";
+  }) { };
+
+  singularity = callPackage (callPackage ./generic.nix rec {
```

Comment:
I've never tried anything like `callPackage (callPackage ...)`

- If we take `.override`, we'll be overriding top-level arguments, i.e. `pname`, etc?
- How do we override nested function's arguments, like `lib`,  `go`, `buildGoModule`, etc?
- Why do we need this?

Reply:
It should be callPackage (import ...). Updated.